makefiles/app_dirs.inc.mk

```
@@ -3,16 +3,55 @@
 # fallback so empty RIOTBASE won't lead to "/examples/"
 RIOTBASE ?= .
 
+# Define the list of examples subdirectories that contain application directories
+EXAMPLES_APPLICATIONS_SUBDIRS :=  \
+    ble                       \
+    coap                      \
+    cord                      \
+    dtls                      \
+    gnrc                      \
+    icn                       \
+    lora                      \
+    mqtt                      \
+    posix                     \
```

```suggestion
EXAMPLES_APPLICATIONS_SUBDIRS :=  $(shell ls -d examples/)
```

Why not just do this? This would prevent something like that.

---

I think the idea is to limit calls to external shell commands as it's less efficient.

---

Ok, then... why not
```suggestion
EXAMPLES_APPLICATIONS_SUBDIRS :=  $(dir $(wildcard examples/*/))
```
...

---

With the proposed solution, you would get unwanted directories listed (in bindist for instance, but not only):

- this PR:

```
$ make info-applications | wc -l
494
```

- your suggestion

```
$ make info-applications | wc -l
496
```

---

Then maybe

`$(filter-out $(dir $(wildcard examples/*/Makefile)),$(wildcard examples/*/))`

to get all subdirectories of examples that do not contain a Makefile (which the grouping dirs would all not)?